### ISSUE TITLE
keyslots are not displayed correctly in omv5

### ISSUE THREAD

**Issue by subzero79** [OpenMediaVault-Plugin-Developers/openmediavault-luksencryption#26]
luksDump changed the output format  
from this
```
Key Slot 0: ENABLED
	Iterations:         	581817
	Salt:               	8b 7f 69 bf 7d fe 43 cc 82 85 6b 05 29 12 c8 45
	                      	ca d3 4e f9 d0 18 ce 0a 58 ce ef 87 24 58 0a 0d
	Key material offset:	8
	AF stripes:            	4000
Key Slot 1: ENABLED
	Iterations:         	558951
	Salt:               	f4 15 c8 32 88 ee 13 8c ce 15 c0 b2 28 89 61 e4
	                      	6e 48 cf 7a 58 1d bd 26 1d c5 9e 00 72 62 c1 ab
	Key material offset:	264
	AF stripes:            	4000
```

to this 

```
Keyslots:
  0: luks2
	Key:        512 bits
	Priority:   normal
	Cipher:     aes-xts-plain64
	Cipher key: 512 bits
	PBKDF:      argon2i
	Time cost:  4
	Memory:     401067
	Threads:    1
	Salt:       54 33 4a 0e a5 02 05 31 a2 40 d0 63 a7 4d 49 92
	            c7 fc 83 66 3e 5d 9d 1a b2 10 ab c9 f2 ec 9d 3e
	AF stripes: 4000
	AF hash:    sha256
	Area offset:32768 [bytes]
	Area length:258048 [bytes]
	Digest ID:  0
  1: luks2
	Key:        512 bits
	Priority:   normal
	Cipher:     aes-xts-plain64
	Cipher key: 512 bits
	PBKDF:      argon2i
	Time cost:  4
	Memory:     369713
	Threads:    1
	Salt:       e5 b8 2d 65 f5 f0 3c d6 97 14 0a dc d5 2a 0c cd
	            d5 f0 b8 62 54 04 6a 79 cb ce d8 96 9d 58 90 68
	AF stripes: 4000
	AF hash:    sha256
	Area offset:290816 [bytes]
	Area length:258048 [bytes]
	Digest ID:  0
```

Now current buster cryptsetup has an option --debug-json which prints the keyslots in json and other info but in the middle of the whole output. Weird

Comes something like this 

```
root@omv5-dev:~# cryptsetup luksDump --debug-json  /dev/vdd
# cryptsetup 2.1.0 processing "cryptsetup luksDump --debug-json /dev/vdd"
# Running command luksDump.
# Locking memory.
# Installing SIGINT/SIGTERM handler.
# Unblocking interruption on signal.
# Allocating context for crypt device /dev/vdd.
# Trying to open and read device /dev/vdd with direct-io.
# Initialising device-mapper backend library.
# Trying to load any crypt type from device /dev/vdd.
# Crypto backend (OpenSSL 1.1.1c  28 May 2019) initialized in cryptsetup library version 2.1.0.
# Detected kernel Linux 5.2.0-0.bpo.2-amd64 x86_64.
# Loading LUKS2 header (repair disabled).
# Opening lock resource file /run/cryptsetup/L_254:48
# Acquiring read lock for device /dev/vdd.
# Verifying read lock handle for device /dev/vdd.
# Device /dev/vdd READ lock taken.
# Trying to read primary LUKS2 header at offset 0x0.
# Opening locked device /dev/vdd
# Veryfing locked device handle (bdev)
# LUKS2 header version 2 of size 16384 bytes, checksum sha256.
# Checksum:8fafc8aaa88f746f036bb3d583f125e360b45789a39a4c93de6992025fce6a0a (on-disk)
# Checksum:8fafc8aaa88f746f036bb3d583f125e360b45789a39a4c93de6992025fce6a0a (in-memory)
# Trying to read secondary LUKS2 header at offset 0x4000.
# Opening locked device /dev/vdd
# Veryfing locked device handle (bdev)
# LUKS2 header version 2 of size 16384 bytes, checksum sha256.
# Checksum:1fd72260cbed3fa97e52c72c5b526ecdd6fc51bcb924ea54e5dac58fcee3212a (on-disk)
# Checksum:1fd72260cbed3fa97e52c72c5b526ecdd6fc51bcb924ea54e5dac58fcee3212a (in-memory)
# Device size 1073741824, offset 16777216.
# Device /dev/vdd READ lock released.
# Only 1 active CPUs detected, PBKDF threads decreased from 4 to 1.
# Not enough physical memory detected, PBKDF max memory decreased from 1048576kB to 505320kB.
# PBKDF argon2i, hash sha256, time_ms 2000 (iterations 0), max_memory_kb 505320, parallel_threads 1.
# {
  "keyslots":{
    "0":{
      "type":"luks2",
      "key_size":64,
      "af":{
        "type":"luks1",
        "stripes":4000,
        "hash":"sha256"
      },
      "area":{
        "type":"raw",
        "offset":"32768",
        "size":"258048",
        "encryption":"aes-xts-plain64",
        "key_size":64
      },
      "kdf":{
        "type":"argon2i",
        "time":4,
        "memory":438162,
        "cpus":1,
        "salt":"OZtstLYVgjGK1VDbPayqLnm8BfkS4z1UwbcBmO5SwcE="
      }
    },
    "1":{
      "type":"luks2",
      "key_size":64,
      "af":{
        "type":"luks1",
        "stripes":4000,
        "hash":"sha256"
      },
      "area":{
        "type":"raw",
        "offset":"290816",
        "size":"258048",
        "encryption":"aes-xts-plain64",
        "key_size":64
      },
      "kdf":{
        "type":"argon2i",
        "time":4,
        "memory":392461,
        "cpus":1,
        "salt":"Ce5bIcPUm2TLRVoe92TJlrdlHSBB505BSeGj2O\/Gpc8="
      }
    }
  },
  "tokens":{
  },
  "segments":{
    "0":{
      "type":"crypt",
      "offset":"16777216",
      "iv_tweak":"0",
      "size":"dynamic",
      "encryption":"aes-xts-plain64",
      "sector_size":512
    }
  },
  "digests":{
    "0":{
      "type":"pbkdf2",
      "keyslots":[
        "0",
        "1"
      ],
      "segments":[
        "0"
      ],
      "hash":"sha256",
      "iterations":87849,
      "salt":"n3WwTa3Of0hn62N64zjMZaQZ\/bB4DQhND8psEMdhRBY=",
      "digest":"tEMuYZulApCeqIGwbhuHAQVWq4ESimZtz3awD2W7GdY="
    }
  },
  "config":{
    "json_size":"12288",
    "keyslots_size":"16744448"
  }
}
LUKS header information
Version:       	2
Epoch:         	4
Metadata area: 	16384 [bytes]
Keyslots area: 	16744448 [bytes]
UUID:          	ada57cad-239d-43da-aee8-6d35c0c88c6c
Label:         	(no label)
Subsystem:     	(no subsystem)
Flags:       	(no flags)

Data segments:
  0: crypt
	offset: 16777216 [bytes]
	length: (whole device)
	cipher: aes-xts-plain64
	sector: 512 [bytes]

Keyslots:
  0: luks2
	Key:        512 bits
	Priority:   normal
	Cipher:     aes-xts-plain64
	Cipher key: 512 bits
	PBKDF:      argon2i
	Time cost:  4
	Memory:     438162
	Threads:    1
	Salt:       39 9b 6c b4 b6 15 82 31 8a d5 50 db 3d ac aa 2e 
	            79 bc 05 f9 12 e3 3d 54 c1 b7 01 98 ee 52 c1 c1 
	AF stripes: 4000
	AF hash:    sha256
	Area offset:32768 [bytes]
	Area length:258048 [bytes]
	Digest ID:  0
  1: luks2
	Key:        512 bits
	Priority:   normal
	Cipher:     aes-xts-plain64
	Cipher key: 512 bits
	PBKDF:      argon2i
	Time cost:  4
	Memory:     392461
	Threads:    1
	Salt:       09 ee 5b 21 c3 d4 9b 64 cb 45 5a 1e f7 64 c9 96 
	            b7 65 1d 20 41 e7 4e 41 49 e1 a3 d8 ef c6 a5 cf 
	AF stripes: 4000
	AF hash:    sha256
	Area offset:290816 [bytes]
	Area length:258048 [bytes]
	Digest ID:  0
Tokens:
Digests:
  0: pbkdf2
	Hash:       sha256
	Iterations: 87849
	Salt:       9f 75 b0 4d ad ce 7f 48 67 eb 63 7a e3 38 cc 65 
	            a4 19 fd b0 78 0d 08 4d 0f ca 6c 10 c7 61 44 16 
	Digest:     b4 43 2e 61 9b a5 02 90 9e a8 81 b0 6e 1b 87 01 
	            05 56 ab 81 12 8a 66 6d cf 76 b0 0f 65 bb 19 d6 
# Releasing crypt device /dev/vdd context.
# Releasing device-mapper backend.
# Unlocking memory.
Command successful.
```

Any ideas on how to get the json output?

I'll leave the issue open

**Comment by subzero79** [comment]
Just to clarify the output is different based on the current luks format. 

LUKS1 will displayed normal and the plugin should be able to parse it, LUKS2 will be displayed in the new format and we need to find a way to parse it.

**Comment by subzero79** [comment]
Closed with #27
